### PR TITLE
Buildkite

### PR DESCRIPTION
**Pull request by jonasgraterol**
Moving from travis to buildkite

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    command: NODE_ENV=development yarn install --frozen-lockfile
+    plugins: *plugins
+
+  - wait
+
+  - label: 'All Tests'
+    command: yarn test
+    plugins: *plugins
+
+  - label: ':rocket: Deploy'
```
Comment (bobzoller) [inline .buildkite/pipeline.yml:18]: this deploy step is for an application, whereas this is a node module.  here are a couple examples of node modules that are already on buildkite to use as an example:

```
meaty~/Projects % ag 'yarn publish' */.buildkite/pipeline.yml
goodeggs-authentication-tokens/.buildkite/pipeline.yml
41:      - yarn publish

goodeggs-kodif-domain-events-adapter-contracts/.buildkite/pipeline.yml
25:      - yarn publish

goodeggs-orders-service-contracts/.buildkite/pipeline.yml
25:      - yarn publish

goodeggs-product-feed-contracts/.buildkite/pipeline.yml
25:      - yarn publish

goodeggs-tsconfig/.buildkite/pipeline.yml
```

Comment (jonasgraterol) [reply .buildkite/pipeline.yml]: Nice, thank you!

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  - label: 'All Tests'
+    command: yarn test
+    plugins: *plugins
+
+  - label: ':loudspeaker: Publish'
+    if: build.tag != null
+    command:
+      - export PATH="/tmp/ci-tools:./node_modules/.bin:$PATH"
+      - yarn publish
```
Comment (bobzoller) [inline .buildkite/pipeline.yml:22]: I'm almost certain you'll need an "automation token" to publish ... the default one available on buildkite is read-only.  I think that means you'll also need an `.npmrc` file that references `NPM_TOKEN`.  I'll give you the value over Slack.

**Review by bobzoller — CHANGES_REQUESTED** [review]
https://github.com/goodeggs/terraform-github/pull/73  will get the buildkite pipelines actually running, and I'll send you an NPM_TOKEN

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    concurrency: 1
+    concurrency_group: 'goodeggs-money/publish'
+    plugins: *plugins
+      secrets:
+        NPM_TOKEN: dZ4gJyTD7zE/V95QfCffWZzQ6SxFW2FA8g/s9Lgh4ckkJIKUUf7rBAHZGrtUdZfT1hZY0IS7RuuTIAQMd-BhTQWNA
```
Comment (bobzoller) [inline .buildkite/pipeline.yml:27]:
```suggestion
    plugins:
      - ssh://git@github.com/goodeggs/goodeggs-core-buildkite-plugin#v0.0.18: *plugin-node-cache
      - ssh://git@github.com/goodeggs/goodeggs-core-buildkite-plugin#v0.0.18:
          secrets:
            NPM_TOKEN: dZ4gJyTD7zE/V95QfCffWZzQ6SxFW2FA8g/s9Lgh4ckkJIKUUf7rBAHZGrtUdZfT1hZY0IS7RuuTIAQMd-BhTQWNA
```

**Review by bobzoller — CHANGES_REQUESTED** [review]

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (jonasgraterol) [reply .buildkite/pipeline.yml]: 👀 ✅

**Review by bobzoller — APPROVED** [review]
LGTM 🤞

**Pull request merged**